src/createRouter.ts

```diff
@@ -121,6 +143,21 @@ const createRouter: RouterFactory = compose<RouterMixin, RouterOptions>({
 		}
 	},
 
+	observeHistory(history: History, context: Context, dispatchInitial: boolean): PausableHandle {
+		if (!historyMap.has(this)) {
+			const listener = pausable(history, 'change', (event: HistoryChangeEvent) => {
+				this.dispatch(context, event.value);
```

Comment:
TS 1.8 infers `this` as `any`, so perhaps add a `const router: Router = this;` at the top and then use `router` rather than `this`.